graphpype/tests/test_utils.py

```diff
@@ -0,0 +1,97 @@
+from graphpype.utils import (get_first, get_second, show_length, show_files,
+                             get_multiple_indexes, random_product,
+                             check_dimensions, check_np_shapes,
+                             check_np_dimension)
+
+import pytest
+import itertools
+
+import numpy as np
+
+two_elem_int_list = [1, 2]
+two_elem_int_tuple = (1, 2)
+
+
+def test_get_first():
+    """test_get_first"""
+    with pytest.raises(ValueError, message="Expecting ValueError"):
+        get_first(1)
+
+    with pytest.raises(ValueError, message="Expecting ValueError"):
+        get_first([])
+
+    assert get_first(two_elem_int_list) == 1
+    assert get_first(two_elem_int_tuple) == 1
+
+
+def test_get_second():
+    """test_get_first"""
+    with pytest.raises(ValueError, message="Expecting ValueError"):
+        get_second([1])
+
+    assert get_second(two_elem_int_list) == 2
+    assert get_second(two_elem_int_tuple) == 2
+
+
+def test_show_length():
+    """test_show_length"""
+    with pytest.raises(ValueError, message="Expecting ValueError"):
+        show_length(1)
```

Comment:
test function with two lines tells me that you don't need a separate function for this. You need at least 10-20 lines in a function :)

Reply:
@jasmainak You are right, function of 2 lines are a bit ridiculous... but I dont know if you have ever used nipype, typically you have "on the fly mecanism" that allows you to make make something WITHIN the connnect function (otherwise you would have to define a full node):

See "Modify Values between Connections":
http://miykael.github.io/nipype-beginner-s-guide/firstSteps.html#"

and you will understand the principles of the functions in utils. :)

Also please MRG when you can, because I want to work on another branch and last time I did before PR was accepted it ended in a mess !

Reply:
sure but still that does not explain why you need to have the functions in `test_utils.py` so small. Anway, let me merge so that you can iterate on top of it.